frontend/src/metabase/dashboard/actions/parameters.ts

```diff
@@ -298,6 +301,13 @@ export const setParameterRequired = createThunkAction(
         required,
       }));
     }
+
+    if (required) {
```

Comment:
shouldn't we track only when parameter becomes required from a not-required state?

Reply:
Do you mean the case when `required` is `true` and `paramater.required` is also true? 
We don't emit this action like that but I could add an additional check if you think it makes sense.

Reply:
I actually mean (if I understand the logic correctly) that we need to track when parameter wasn't required and then became required

Reply:
Yeah, I understood. In case we were emitting this action without checks (now because it's a "switch" you can't emit it when a parameter is already required), it would make sense. B broadly speaking, I don't think this is unnecessary. 

This event is used to track feature adoption, which Conor agreed doesn't have to be very precise. That's why we didn't track it at saving for instance and instead track just toggling it on.